### ISSUE TITLE
Change in @AutoConfigureStubRunner behaviour in Spring Cloud Contract 2.0.2

### ISSUE THREAD

**Issue by davidmelia** [spring-cloud/spring-cloud-contract#809]
Hi,

I'm using Spring Boot 2.0.7.RELEASE, Cloud Finchley.SR1 (Contract 2.0.1) with Cucumber 4.2.0 where my cucumber steps are annotated with:

`@RunWith(SpringRunner.class)
@SpringBootTest(classes = Application.class, webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
@ActiveProfiles({"cucumber-test"})
@AutoConfigureStubRunner(stubsMode = StubsMode.LOCAL, ids = {"uk.co.service:some-service:+:stubs:6980"}
public class CucumberStep {
}
`

Where all tests are running fine and fast.

After upgrade to Finchley.SR2 (Contract 2.0.2) there has been a change in behaviour of @AutoConfigureStubRunner  where by I get

`org.springframework.web.client.ResourceAccessException: I/O error on PUT request for "http://localhost:6980/some-service/api/2/xyz": localhost:6980 failed to respond; nested exception is org.apache.http.NoHttpResponseException: localhost:6980 failed to respond`
aka my stub hasn't started before the test has executed. 

The failures seem to happen with tests running in the same spring context and if I put a breakpoint in the new class StubRunnerWireMockTestExecutionListener.beforeTestClass then the problem goes away i.e. it's a race condition issue.  Further to this I can also get around this by either adding DirtiesContext to my CucumberStep class (adds minutes to my test run as I have quite a few tests!!) or Thread.sleep(2000) before each test which sucks.

Any ideas?

Thanks

**Comment by OlgaMaciaszek** [comment]
@davidmelia thanks for reporting it - we have not come across such bug before. Could you please provide a sample demo project that reproduces this error (a link to a git repo or zip) so that we can verify it?

**Comment by davidmelia** [comment]
Yes I will - might take me a few days-ish to detangle this issue from my code base.  Will get this to you as soon as I can.

**Comment by OlgaMaciaszek** [comment]
Thanks @davidmelia - waiting for it.

**Comment by davidmelia** [comment]
Just to confirm this issue affects Greenwich.RC1 as well.  

Thanks
